x/wasm/client/cli/genesis_msg.go

```diff
@@ -293,6 +284,52 @@ func getAllCodes(state *types.GenesisState) ([]CodeMeta, error) {
 	return all, nil
 }
 
+type contractMeta struct {
+	ContractAddress string             `json:"contract_address"`
+	Info            types.ContractInfo `json:"info"`
+}
+
+func getAllContracts(state *types.GenesisState) []contractMeta {
+	var all []contractMeta
+	for _, c := range state.Contracts {
+		all = append(all, contractMeta{
+			ContractAddress: c.ContractAddress,
+			Info:            c.ContractInfo,
+		})
+	}
+	// add inflight
+	seq := contractSeqValue(state)
+	for _, m := range state.GenMsgs {
+		if msg := m.GetInstantiateContract(); msg != nil {
+			all = append(all, contractMeta{
+				ContractAddress: contractAddress(msg.CodeID, seq).String(),
+				Info: types.ContractInfo{
+					CodeID:  msg.CodeID,
+					Creator: msg.Sender,
+					Admin:   msg.Admin,
+					Label:   msg.Label,
+				},
+			})
+			seq++
+		}
+	}
+	return all
+}
+
+func hasAccountBalance(cmd *cobra.Command, a map[string]json.RawMessage, sender sdk.AccAddress, coins sdk.Coins) (bool, error) {
```

Comment:
variable `a` is a bit confusing.

But I would note this check is only required if `! coins.IsEmpty()`. I guess it could make sense that funded only genesis accounts send messages, but if we just want to prevent obvious errors, we can short circuit this is we are not sending any balance, so we let accounts with no balances create contracts as long as they don't interact with funds.